### DISCUSSION TITLE
Relations only return ids (even with expand).

### DISCUSSION THREAD

**Discussion by Amir-Zouerami** [pocketbase/pocketbase#3758] [Q&A] [answered]
I have a `posts` collection which has many columns. Two of them are relation fields and are called `post_categories` (keeps some metadata about posts in another collection called `categories`) and `comments` (keeps comments in another collection called `comments`).

![image](https://github.com/pocketbase/pocketbase/assets/53701884/020d940d-e92b-4c0d-9bb6-710fb25b6869)

I use this code to get a post with the categories for example:

```ts
    const pb = new Pocketbase(process.env.PB_DOMAIN);
    await pb.admins.authWithPassword(
      process.env.PB_ADMIN_EM as string,
      process.env.PB_ADMIN_PS as string
    );

    const post = await pb
      .collection('posts')
      .getOne('123456789456789', { expand: 'post_categories' });

    console.log(post);
```

What I get from this is only the `id`s of my relations:

```
// rest of the data

  cover: 'sample_blog_post_cover_main_page_DVpzfRZH0p.webp',
  created: '2023-11-03 08:04:23.952Z',
  expand: { post_categories: [ [Object], [Object] ] },
  id: '123456789456789',

// rest of the data
```
`post_categories` only contain the id and no other columns. If I query without `expand` I'll get the same thing:

```
// rest of the data

  cover: 'sample_blog_post_cover_main_page_DVpzfRZH0p.webp',
  created: '2023-11-03 08:04:23.952Z',
  id: '123456789456789',
  post_categories: [ '123456789012347', '123456789012346' ],

// rest of the data
```

How can I get the actual data and not just `id`s? Am I missing something here?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure I follow. The 2 responses are not the same.
The expanded relation is located under the `expand.post_categories` array of objects in your first response.

**Reply by Amir-Zouerami** [reply]
Never mind! Figured it out. My script was the problem.
